test/stringio/test_stringio.rb

```diff
@@ -990,7 +990,8 @@ def test_coderange_after_overwrite
     assert_predicate(s.string, :ascii_only?)
   end
 
-  if eval(%{ "test".frozen? && !"test".equal?("test") }) # Ruby 3.4+ chilled strings
+  require "objspace"
+  if ObjectSpace.dump(eval(%{"test"})).include?('"chilled":true') # Ruby 3.4+ chilled strings
```

Comment:
This works fine on CRuby and TruffleRuby but fails on JRuby:
https://github.com/ruby/stringio/actions/runs/9302740210/job/25603570664?pr=96#step:9:8

That's continue-on-error I guess because it's failing for different reasons on master: https://github.com/ruby/stringio/actions/runs/8714902878/job/23905912842